packages/hardhat-zksync-node/src/utils.ts

```diff
@@ -171,27 +178,52 @@ export async function getLatestRelease(owner: string, repo: string, userAgent: s
         },
     });
 
-    // Check if the response is a redirect
-    if (response.statusCode >= 300 && response.statusCode < 400) {
-        // Get the URL from the 'location' header
-        if (response.headers.location && typeof response.headers.location === 'string') {
-            // Check if the redirect URL matches the expected pattern
-            if (response.headers.location.startsWith(redirectUrlPattern)) {
-                // Extract the tag from the redirect URL
-                return response.headers.location.substring(redirectUrlPattern.length);
-            }
+    if (response.statusCode === 200) {
+        return JSON.parse(await response.body.text()).map((tag: any) => tag.name.slice(1));
+    } else {
+        throw new ZkSyncNodePluginError(`Unexpected response status: ${response.statusCode} for URL: ${finalUrl}`);
+    }
+}
+
+async function handleRedirect(url: string, userAgent: string, timeout: number): Promise<string> {
+    const { request } = await import('undici');
+
+    let currentUrl = url;
 
-            throw new ZkSyncNodePluginError(`Unexpected redirect URL: ${response.headers.location} for URL: ${url}`);
+    while (true) {
+        const response = await request(currentUrl, {
+            headersTimeout: timeout,
+            maxRedirections: 0,
+            method: 'GET',
+            headers: {
+                'User-Agent': `${userAgent}`,
+            },
+        });
+
+        if (response.statusCode >= 300 && response.statusCode < 400) {
+            if (response.headers.location && typeof response.headers.location === 'string') {
+                currentUrl = response.headers.location;
+                continue;
+            } else {
+                throw new ZkSyncNodePluginError(`Redirect location not found for URL: ${currentUrl}`);
+            }
         } else {
-            // Throw an error if the 'location' header is missing in a redirect response
-            throw new ZkSyncNodePluginError(`Redirect location not found for URL: ${url}`);
+            return currentUrl;
         }
-    } else {
-        // Throw an error for non-redirect responses
-        throw new ZkSyncNodePluginError(`Unexpected response status: ${response.statusCode} for URL: ${url}`);
     }
 }
 
+export function resolveTag(tags: string[], latestTag: string, initialTag: string): string {
+    if (initialTag === 'latest') {
+        return latestTag;
+    }
+    const [major, minor, patch] = initialTag.split('.');
+    const tag = tags.find((t) => t.startsWith(patch === '*' ? `${major}.${minor}` : initialTag));
```

Comment:
Can we rely on tags sorted here so the first match for patch `*` is always the latest patch?